automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java

```diff
@@ -0,0 +1,154 @@
+            }
+
+            inputStream = new FileInputStream(scenarioResultFile);
+            XMLEventReader eventReader = factory.createXMLEventReader(inputStream);
+            TestCase testCase = null;
+            while (eventReader.hasNext()) {
+                XMLEvent event = eventReader.nextEvent();
+                switch(event.getEventType()) {
+                    case XMLStreamConstants.START_ELEMENT:
+                        StartElement startElement = event.asStartElement();
+                        String elementName = startElement.getName().getLocalPart();
+                        if (HTTP_SAMPLE_ELEMENT.equalsIgnoreCase(elementName) ||
+                                SAMPLE_ELEMENT.equalsIgnoreCase(elementName)) {
+                            testCase = this.addAttributes(startElement);
+                        } else if (FAILURE_MESSAGE_ELEMENT.equalsIgnoreCase(elementName)) {
+                            failureMsgElement = true;
+                        }
+                        break;
+                    case XMLStreamConstants.CHARACTERS:
+                        Characters characters = event.asCharacters();
+                        if (failureMsgElement) {
+                            testCase.setFailureMessage(characters.getData());
+                            failureMsgElement = false;
+                        }
+                        break;
+                    case XMLStreamConstants.END_ELEMENT:
+                        EndElement endElement = event.asEndElement();
+                        String nodeName = endElement.getName().getLocalPart();
+                        if (HTTP_SAMPLE_ELEMENT.equalsIgnoreCase(nodeName) ||
+                                SAMPLE_ELEMENT.equalsIgnoreCase(nodeName)) {
+                            this.testScenario.addTestCase(testCase);
+                        }
+                        break;
+                }
+            }
+
+            if (logger.isDebugEnabled()) {
+                logger.debug("End parsing scenario-results file of the TestScenario : '" + testScenarioName +
+                        "' using the FunctionalTestResultParser");
+            }
+        } catch (XMLStreamException e) {
+            throw new JMeterResultParserException("Unable to parse the scenario-results file of TestScenario :" +
+                    testScenarioName, e);
+        } catch (FileNotFoundException e) {
+            throw new JMeterResultParserException("Unable to locate the scenario-results file.", e);
+        } finally {
+            try {
+                if (inputStream != null) {
+                    inputStream.close();
+                }
+            } catch (IOException e) {
+                throw new JMeterResultParserException("Unable to close the input stream of scenario results file of " +
+                        "the TestScenario : " + testScenarioName, e);
+            }
+        }
+    }
+
+    private TestCase addAttributes(StartElement startElement) {
```

Comment:
Missing javadocs.
How about we change the method name from addAttributes to something more meaningful? Say, buildTestCaseFrom? The input parameter name can be something like jmeterSampleElement. wdyt?